src/Clients/MainApp/FSpot/ThumbnailCache.cs

```diff
@@ -127,43 +125,28 @@ public void Dispose ()
 
 		protected virtual void Dispose(bool disposing)
 		{
-			if (disposing)
-			{
-				foreach (var item in pixbuf_mru) {
-					Thumbnail thumb = item;
-					pixbuf_hash.Remove (thumb.uri);
-					thumb.pixbuf.Dispose ();
-				}
-				pixbuf_mru.Clear ();
+			foreach (var thumb in pixbuf_mru) {
+				pixbuf_hash.Remove (thumb.Uri);
+				thumb.Pixbuf.Dispose ();
 			}
-			else
-			{
-				foreach (var item in pixbuf_mru) {
-					Thumbnail thumb = item;
-					pixbuf_hash.Remove (thumb.uri);
-					thumb.pixbuf.Dispose ();
-				}
-				pixbuf_mru.Clear ();
-			} 
-
+			pixbuf_mru.Clear ();
 		}
 
 		~ThumbnailCache ()
 		{
-			Console.WriteLine ("Finalizer called on {0}. Should be Disposed", GetType ());
 			Dispose (false);
```

Comment:
Dispose doesn't use 'disposing' anymore. Do we even need this? Or should 'disposing' be used?